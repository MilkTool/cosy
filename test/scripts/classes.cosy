class Breakfast {
    cook() {
        print "Eggs a-fryin'!"
    }

    serve(who) {
        print "Enjoy your breakfast, " + who + "."
    }

    tasteCake() {
        var adjective = "delicious"
        print "The " + this.cakeFlavor + " cake is " + adjective + "!"
    }
}

print Breakfast

var breakfast = Breakfast() // tests instancing
print breakfast

breakfast.eggs = true // tests instance setter
breakfast.eggCount = 3
if (breakfast.eggs) { // tests instance getter
    print "You have choosen to have " + breakfast.eggCount + " eggs"
}
else if (!breakfast.eggs) print "You have choosen to not have eggs"

breakfast.cook() // tests method call

var speakTo = breakfast.serve
speakTo("Mr. Bond") // tests reference to instance method

breakfast.cakeFlavor = "German chocolate"
breakfast.tasteCake() // tests 'this' instance
