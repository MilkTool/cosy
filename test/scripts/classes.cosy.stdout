Breakfast
Breakfast instance
You have choosen to have 3 eggs
Eggs a-fryin'!
Enjoy your breakfast, Mr. Bond.
The German chocolate cake is delicious!
